com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/model/ServiceInstanceWithServiceKeys.java

```diff
@@ -0,0 +1,26 @@
+package com.sap.cloud.lm.sl.cf.core.model;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.cloudfoundry.client.lib.domain.CloudServiceInstance;
+import org.cloudfoundry.client.lib.domain.CloudServiceKey;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface ServiceInstanceWithServiceKeys {
```

Comment:
What is the benefit of having this class? As far as I can see, it is used only in delete services step, do you think it'd be simpler to just acquire the service and the service keys as separate entities and process them accordingly. What does this 'wrapping' help with? Wouldn't the code be more understandable and explicit in DeleteServicesStep if that was applied? What do you think